### PR TITLE
PCC02 ryanh153

### PR DESCRIPTION
**Pull request by ryanh153**
ATTENTION: before clicking "Create Pull Request" please submit some meta data, thanks!

Difficulty level (1-10): []
Estimated time spent (hours): []
Completed (yes/no): []
I stretched my coding skills (if yes what did you learn?): []
Other feedback (what can we improve?): []

### CODE REVIEW

**Comment by ryanh153** [comment]
Difficulty level (1-10): [5]
Estimated time spent (hours): [2]
Completed (yes/no): [Yes]
I stretched my coding skills (if yes what did you learn?): [Permutations and practice with recursive functions is always good.]
Other feedback (what can we improve?): []

File: 02/ryanh153/game.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    """Calc a given word value based on Scrabble LETTER_SCORES mapping"""
+    return sum(LETTER_SCORES.get(char.upper(), 0) for char in word)
+
+
+# re-use from challenge 01
+def max_word_value(words):
+    """Calc the max value of a collection of words"""
+    return max(words, key=calc_word_value)
+
+# check if user word is valid
```
Comment (bbelderbos) [inline 02/ryanh153/game.py]: docstring?

File: 02/ryanh153/game.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+		currWord = max_word_value(possibleWords)
+		currScore = calc_word_value(currWord)
+	else:
+		currScore = 0
+
+	if currScore > maxScore:
+		maxScore = currScore
+		maxLetters = currWord
+
+	if len(letters) > 0:
```
Comment (bbelderbos) [inline 02/ryanh153/game.py]: seems complex, refactor?

File: 02/ryanh153/game.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			break
+
+	if word.lower() not in DICTIONARY:
+		valid = False
+
+	return valid
+
+# get all possible words from the given list letters
+def getBest(letters, maxScore, maxLetters):
+	possibleWords = ["".join(word) for word in permutations(letters) if "".join(word).lower() in DICTIONARY]
```
Comment (bbelderbos) [inline 02/ryanh153/game.py]: 2x join

**Pull request merged**